code/modules/research/techweb/_techweb.dm

```diff
@@ -0,0 +1,253 @@
+
+/proc/initialize_all_techweb_nodes(clearall = FALSE)
+	if(islist(GLOB.techweb_nodes) && clearall)
+		QDEL_LIST(GLOB.techweb_nodes)
+	if(islist(GLOB.techweb_nodes_starting && clearall))
+		QDEL_LIST(GLOB.techweb_nodes_starting)
+	var/list/returned = list()
+	for(var/path in typesof(/datum/techweb_node))
+		var/datum/techweb_node/TN = path
+		if(isnull(initial(TN.id)))
+			continue
+		TN = new path
+		returned[initial(TN.id)] = TN
+		if(TN.starting_node)
+			GLOB.techweb_nodes_starting[TN.id] = TN
+	GLOB.techweb_nodes = returned
+	calculate_techweb_nodes()
+
+/proc/initialize_all_techweb_designs(clearall = FALSE)
+	if(islist(GLOB.techweb_designs) && clearall)
+		QDEL_LIST(GLOB.techweb_designs)
+	var/list/returned = list()
+	for(var/path in typesof(/datum/design))
+		var/datum/design/DN = path
+		if(isnull(initial(DN.id)))
+			continue
+		DN = new path
+		returned[initial(DN.id)] = DN
+	GLOB.techweb_designs = returned
+
+/proc/get_techweb_node_by_id(id)
+	if(GLOB.techweb_nodes[id])
+		return GLOB.techweb_nodes[id]
+
+/proc/get_techweb_design_by_id(id)
+	if(GLOB.techweb_designs[id])
+		return GLOB.techweb_designs[id]

```

Comment:
This could just be return GLOB.techweb_designs[id]. If that would be null, you just return null as you should.